pulp_service/pulp_service/tests/functional/test_domain_based_permissions.py

```diff
@@ -111,3 +110,126 @@ def test_user_list_domain_permissions(pulpcore_bindings, anonymous_user):
         )
 
         pulpcore_bindings.DomainsApi.list()
+
+
+def test_only_owners_can_delete_domain(pulpcore_bindings, anonymous_user, gen_object_with_cleanup):
+    user1_orgid1 = {
+        "identity": {
+            "internal": {
+                "org_id": 1
+            },
+            "user": {
+                "username": "user1"
+            }
+        }
+    }
+
+    user2_orgid2 = {
+        "identity": {
+            "internal": {
+                "org_id": 2
+            },
+            "user": {
+                "username": "user2"
+            }
+        }
+    }
+
+    with anonymous_user:
+        # Clear any authentication header
+        pulpcore_bindings.DomainsApi.api_client.default_headers.pop(
+            "x-rh-identity", None
+        )
+
+        header_user1_orgid1 = json.dumps(user1_orgid1)
+        auth_user1_orgid1 = b64encode(bytes(header_user1_orgid1, "ascii"))
+
+        pulpcore_bindings.DomainsApi.api_client.default_headers["x-rh-identity"] = (
+            auth_user1_orgid1
+        )
+
+        # User 1 creates a domain
+        domain_name = str(uuid4())
+        domain = gen_object_with_cleanup(
+            pulpcore_bindings.DomainsApi, {
+                "name": domain_name,
+                "storage_class": "pulpcore.app.models.storage.FileSystem",
+                "storage_settings": {"MEDIA_ROOT": "/var/lib/pulp/media/"},
+            }
+        )
+
+        # User 2 tries to delete the domain
+        header_user2_orgid2 = json.dumps(user2_orgid2)
+        auth_user2_orgid2 = b64encode(bytes(header_user2_orgid2, "ascii"))
+
+        pulpcore_bindings.DomainsApi.api_client.default_headers["x-rh-identity"] = (
+            auth_user2_orgid2
+        )
+
+        with pytest.raises(pulpcore_bindings.ApiException) as exp:
+            pulpcore_bindings.DomainsApi.delete(domain.pulp_href)
+
+        assert exp.value.status == 403
```

Comment:
can you also test that the user1 can delete their own domain?

Reply:
Done